ls1intum/Artemis
# `Development`: Remove unused knowledge functionality

### Checklist
#### General


#### Server

### Motivation and Context
* I removed unused functionality which is not needed anymore. This simplifies code
* I made the deletion of text and modeling exercises more consistent
* Additionally, this PR improves server test quality by automatically converting search results instead of using Linked HashMaps.
* I prepared some annotations for the Spring Boot 3 / Hibernate 6 update (in particular, DiscriminatorOptions are not needed)

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Create a text and modeling exercise (course)
2. Participate in the new exercises and check that everything works as usual
3. Import a text and modeling exercise (course)
4. Participate in the imported exercises and check that everything works as usual
5. Delete the exercises and make sure it works without issues

#### Exam Mode Testing



Prerequisites:
- 1 Instructor
- 1 Student

1. Create a text and modeling exercise (exam)
2. Participate in the new exercises and check that everything works as usual
3. Import a text and modeling exercise (exam)
4. Participate in the imported exercises and check that everything works as usual
5. Delete the exercises or the exam and make sure it works without issues

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage
no actual changes as mostly code was removed
